tests/Data/BlockChainData/Test_BlockChain.cpp

```diff
@@ -0,0 +1,251 @@
+    BOOST_CHECK_MESSAGE(blc.AddBlockLink(index, dsindex, blocktype, blockhash),
+                        "Cannot add block link\n");
+  }
+}
+
+bool operator==(const BlockLink& c1, const BlockLink& c2) {
+  return (get<0>(c1) == get<0>(c2) && get<1>(c1) == get<1>(c2) &&
+          get<2>(c1) == get<2>(c2) && get<3>(c1) == get<3>(c2) &&
+          get<4>(c1) == get<4>(c2));
+}
+
+BOOST_AUTO_TEST_CASE(BlockLinkChain_test) {
+  INIT_STDOUT_LOGGER();
+
+  LOG_MARKER();
+
+  BlockLinkChain blc;
+
+  vector<BlockLink> blTest_v;
+  uint64_t blocknum = 1;
+  // Add first BlockLink with index higher than 0
+  BOOST_CHECK_MESSAGE(blc.AddBlockLink(1, 1, DS, BlockHash::random()) == false,
+                      "Can add first BlockLink with index greater than zero " +
+                          to_string(blocknum) + ".\n");
+
+  appendBlockLinkAndChain_v(blc, blTest_v, 1, BLOCKCHAIN_SIZE);
+
+  // Get and compare added random BlockLink
+  uint8_t randBlockLinkIndex =
+      TestUtils::RandomIntInRng<uint8_t>(0, blTest_v.size() - 1);
+  BOOST_CHECK_MESSAGE(
+      blc.GetBlockLink(randBlockLinkIndex) == blTest_v[randBlockLinkIndex],
+      "BlockLink in BlockLinkChain not equals to added one.\n");
+
+  // Access out of index
+  uint64_t index_out = blTest_v.size();
+  BlockLink empty_blc;
+  BOOST_CHECK_MESSAGE(
+      empty_blc == blc.GetBlockLink(index_out),
+      "Empty BlockLinkChain had to be returned when accessed out of index " +
+          to_string(index_out) + ".\n");
+
+  // Get BlockLink from persistent storage - BLOCKCHAIN_SIZE exceeded and
+  // element on index > BLOCKCHAIN_SIZE will be accessed
+  appendBlockLinkAndChain_v(blc, blTest_v, BLOCKCHAIN_SIZE - blTest_v.size(),
+                            BLOCKCHAIN_SIZE + 10);
+  BOOST_CHECK_MESSAGE(
+      blTest_v[1] == blc.GetBlockLink(1),
+      "Incorrect BlockLink returned from persistent storage.\n");
+
+  // Get latest BlockLink
+  BOOST_CHECK_MESSAGE(blc.GetLatestBlockLink() == blTest_v.back(),
+                      "Incorrect latest BlockLink returned.\n");
+
+  // Set and get DSComm
+  PubKey pk_in = TestUtils::GenerateRandomPubKey();
+  Peer peer_in = TestUtils::GenerateRandomPeer();
+  std::deque<std::pair<PubKey, Peer>> dq_in = {make_pair(pk_in, peer_in)};
+  blc.SetBuiltDSComm(dq_in);
+  std::deque<std::pair<PubKey, Peer>> dq_out;
+
+  dq_out = blc.GetBuiltDSComm();
+  PubKey pk_out = dq_out.back().first;
+  Peer peer_out = dq_out.back().second;
+  BOOST_CHECK_MESSAGE(pk_in == pk_out && peer_in == peer_out,
+                      "DSComm obtained not equal to the set one.\n");
+
+  // Add BlockLink with lower index than the latest
+  uint64_t index_old = 1;
+  BOOST_CHECK_MESSAGE(
+      blc.AddBlockLink(index_old, 1, DS, BlockHash::random()) == false,
+      "Can add BlockLink with index " + to_string(index_old) +
+          " lower then the latest index " + to_string(blc.GetLatestIndex()) +
+          ".\n");
+}
+
+DSBlockHeader createDSBlockHeader(const uint64_t& blockNum) {
```

Comment:
If we can reuse this, perhaps we can move it to libTestUtils?
We have an existing `GenerateRandomDSBlockHeader()` there, but not one with a `blockNum` parameter.
You can probably extend that one.

Reply:
Done